boa_ast/src/expression/literal/mod.rs

```diff
@@ -110,6 +110,16 @@ pub enum Literal {
     /// [spec]: https://tc39.es/ecma262/#sec-null-value
     /// [mdn]: https://developer.mozilla.org/en-US/docs/Glossary/null
     Null,
+
+    /// TODO: doc
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-undefined-value
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Glossary/undefined
+    Undefined,
```

Comment:
Uh we had to remove this because `undefined` is a valid identifier e.g. this compiles and prints `5`:
```javascript
{
    let undefined = 5;
    console.log(undefined); // 5
}
```

Reply:
We'll have to remember that this can only be used for optimization purposes and not for parsing the `undefined` token. Can you add a comment explaining it?